constraint/pkg/client/client.go

```diff
@@ -227,12 +227,34 @@ func (c *Client) createTemplateArtifacts(templ *templates.ConstraintTemplate) (*
 		return nil, err
 	}
 
+	entryPointPath := createTemplatePath(targetHandler.GetName(), templ.Spec.CRD.Spec.Names.Kind)
+
+	return &basicCTArtifacts{
+		template:      templ,
+		crd:           crd,
+		targetHandler: targetHandler,
+		namePrefix:    entryPointPath,
+	}, nil
+}
+
+// createTemplateArtifacts will validate the CT, create the CRD for the CT's constraints, then
+// validate and rewrite the rego sources specified in the CT.
+func (c *Client) createTemplateArtifacts(templ *templates.ConstraintTemplate) (*ctArtifacts, error) {
+	artifacts, err := c.createBasicTemplateArtifacts(templ)
+	if err != nil {
+		return nil, err
+	}
+	targetSpec, targetHandler, err := c.validateTargets(templ)
```

Comment:
nit: If we also save `targetSpec` in `basicCTArtifacts` then we can skip this call and use `artifacts.targetSpec`

Reply:
added